lightning/src/ln/onion_route_tests.rs

```diff
@@ -477,11 +477,19 @@ fn test_onion_failure() {
 	bogus_route.paths[0][route_len-1].fee_msat = amt_to_forward;
 	run_onion_failure_test("amount_below_minimum", 0, &nodes, &bogus_route, &payment_hash, &payment_secret, |_| {}, ||{}, true, Some(UPDATE|11), Some(NetworkUpdate::ChannelUpdateMessage{msg: ChannelUpdate::dummy(short_channel_id)}), Some(short_channel_id));
 
+	for node in nodes.iter() {
+		*node.keys_manager.override_session_priv.lock().unwrap() = Some([4; 32]);
+	}
```

Comment:
Seems pre-existing but as you mention these parts are a bit weird/could use comments at some point

Reply:
Yeah, agreed! Mostly looking for a better way to do this rather than the hack here. The problem is the previous test cases use the same `payment_id` because of the earlier override. So we end up getting a pending outbound payment with `payment_hash` rather than the one generated inside `send_along_route`. Maybe there's a way to write these tests without needing the override?

Reply:
FYI, I ended up adding a test-only `clear_pending_payments` method to call instead of using this hack.